### PR TITLE
coding added for  hamburger menu and icon in   css and js fils

### PR DESCRIPTION
**Pull request by Moon2024-light**
hi,team "coding added for  hamburger menu and icon in html,  css and js fils" please review thanks

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -6,6 +6,7 @@
     <title>Document</title>
     <link rel="stylesheet" href="styles/general.css">
     <link rel="stylesheet" href="styles/variables.css">
+    <link rel="stylesheet" href="./styles/index.css">
```
Comment (jasminedns) [inline index.html]: instead of doing the css in main copy paste it in general cause we're gonna use it for every page

File: index.html [hunk 2]
```diff
@@ -17,8 +18,27 @@
...
+    <div class="hamburger-menu" id="hamburgerMenu">
+        <div class="line"></div>
+        <div class="line"></div>
+        <div class="line"></div>
+    </div>
     </header>
 
+    <!-- Side Menu -->
+    <div class="side-menu" id="sideMenu">
+        <ul>
```
Comment (jasminedns) [inline index.html]: You can put the html links in

File: index.html [hunk 3]
```diff
@@ -17,8 +18,27 @@
...
+            <li><a href="#">HOME PAGE</a></li>
```
Comment (jasminedns) [inline index.html]: I'd name this one HOME instead of HOME PAGE

File: styles/index.css [hunk 4]
```diff
@@ -0,0 +1,92 @@
+* 
+body {
+    margin: 0;
+    font-family: 'Arial', sans-serif;
```
Comment (jasminedns) [inline styles/index.css]: we're gonna use koho for this so link the css koho file in the head and change "arial" to "Ko-Ho"

File: styles/index.css [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+    background-color: #F5F0D7; /* Beige background */
```
Comment (jasminedns) [inline styles/index.css]: you can use the variable here

**Review by jasminedns — CHANGES_REQUESTED** [review]
I've commented on some lines where you can edit or add some things. Also remember to use variables so that if we want to change colors or sizes we can do it only in variables and the changes will be applied to every document

**Review by Moon2024-light — COMMENTED** [review]
i made changes please review thanks

File: index.html [hunk 6]
```diff
@@ -6,6 +6,8 @@
     <title>Document</title>
     <link rel="stylesheet" href="styles/general.css">
     <link rel="stylesheet" href="styles/variables.css">
+    <link rel="stylesheet" href="./styles/Fonts/KoHo_WOFF/KohoFonts.css">
+    <link rel="stylesheet" href="./styles/general.css">
```
Comment (jasminedns) [inline index.html:9]: delete this line since general it's already been linked in L7

File: index.html [hunk 7]
```diff
@@ -17,8 +19,27 @@
...
+        <div class="line"></div>
+        <div class="line"></div>
+        <div class="line"></div>
+    </div>
     </header>
 
+    <!-- Side Menu -->
+    <div class="side-menu" id="sideMenu">
+        <ul>
+            <li><a href="#home">HOME</a></li>
```
Comment (jasminedns) [inline index.html]: #home is not a link. You need to put the path to the html page

File: styles/general.css [hunk 8]
```diff
@@ -40,4 +40,97 @@ header {
...
     font-family: 'ZillaSlab';
-}
\ No newline at end of file
+}
+
+
+body {
+    margin: 0;
+    font-family: 'KoHo';
+    --title-tertiary: #F5F0D7;
```
Comment (jasminedns) [inline styles/general.css]: wrong way to use variable

**Review by jasminedns — CHANGES_REQUESTED** [review]
make sure every color and size is using a variable too

File: styles/general.css (hunk 8 above)
Comment (TobiasKjernell) [reply styles/general.css]: var(--title-tertiary);

File: index.html [hunk 9]
```diff
@@ -17,8 +19,27 @@
...
+            <li><a href="#birds">BIRDS</a></li>
+            <li><a href="#reptiles">REPTILES</a></li>
+            <li><a href="#mammals">MAMMALS</a></li>
+            <li><a href="#about-us">ABOUT US</a></li>
+        </ul>
+    </div>
+
+    <script src="./scripts/index.js"></script>
```
Comment (jasminedns) [inline index.html]: also this link has to be in the head and you need to add defer after the source

**Review by jasminedns — CHANGES_REQUESTED** [review]

**Comment by Moon2024-light** [comment]
i made changes please review thanks

File: styles/general.css [hunk 10]
```diff
@@ -40,4 +40,99 @@ header {
...
+    margin: 0;
+    font-family: 'KoHo';
+    color: var(--textcolor-main); /* Using main text color */
+    background-color: var(--textcolor-secondary); /* Background set to secondary text color */
+}
+
+/* Hamburger Menu Icon */
+.hamburger-menu {
+    position: absolute;
+    top: var(--border-radius);
```
Comment (TobiasKjernell) [inline styles/general.css]: don't see why we use the radius variable for something like this. Should have their own if needed.

File: styles/general.css [hunk 11]
```diff
@@ -40,4 +40,99 @@ header {
...
+    border-radius: var(--border-radius);
+    box-shadow: 0px 4px 6px var(--shadow-light); /* Using light shadow */
+}
+
+.hamburger-menu .line {
+    width: 30px;
+    height: 4px;
+    background-color: var(--title-main); /* Default green */
+    margin: calc(var(--border-radius) / 2) 0; /* Using calculated spacing */
+    transition: background-color var(--transition-speed) ease;
```
Comment (TobiasKjernell) [inline styles/general.css]: this variable doesn't exist, need to make this one

File: styles/general.css [hunk 12]
```diff
@@ -40,4 +40,99 @@ header {
...
+.side-menu {
+    position: fixed;
+    top: 0;
+    right: 0;
+    width: 100%;
+    height: 100%;
+    background-color: var(--title-tertiary); /* Beige background */
+    transition: opacity var(--transition-speed) ease-in-out, visibility var(--transition-speed);
+    padding-top: calc(var(--border-radius) * 5);
+    box-shadow: -2px 0 5px var(--shadow-heavy);
```
Comment (TobiasKjernell) [inline styles/general.css]: same here, this variable doesnt exist, make one

**Review by TobiasKjernell — CHANGES_REQUESTED** [review]

File: index.html [hunk 13]
```diff
@@ -17,8 +19,25 @@
...
     </header>
 
+    <!-- Side Menu -->
+    <div class="side-menu" id="sideMenu">
+        <ul>
+            <li><a href="index.html">HOME</a></li> 
+            <li><a href="birds.html">BIRDS</a></li> 
+            <li><a href="reptiles.html">REPTILES</a></li> 
+            <li><a href="mammals.html">MAMMALS</a></li> 
+            <li><a href="about.html">ABOUT US</a></li>
```
Comment (jasminedns) [inline index.html]: the index link is right but the rest need to go inside the html folder first ex. for birds would be: href="html/birds.html"

File: styles/general.css [hunk 14]
```diff
@@ -40,4 +40,99 @@ header {
...
+/* Hamburger Menu Icon */
+.hamburger-menu {
+    position: absolute;
+    top: var(--border-radius);
+    right: var(--border-radius);
+    cursor: pointer;
+    z-index: 1000;
+    padding: var(--border-radius);
+    border-radius: var(--border-radius);
+    box-shadow: 0px 4px 6px var(--shadow-light); /* Using light shadow */
```
Comment (jasminedns) [inline styles/general.css]: the variable does not exist at the moment, so it won't know what you mean. You can just type in the value you want to or create a new variable in variables.css with the same name and give it a value

File: styles/general.css (hunk 10 above)
Comment (jasminedns) [inline styles/general.css]: use the border radius variable for border radius only

**Review by jasminedns — CHANGES_REQUESTED** [review]

**Review by jasminedns — CHANGES_REQUESTED** [review]
added some new comment reviews

**Review by Moon2024-light — COMMENTED** [review]
please review my changes thanks

File: styles/general.css [hunk 15]
```diff
@@ -40,4 +40,95 @@ header {
...
 .titleWrapper h1, .titleWrapper h2 {
     font-family: 'ZillaSlab';
-}
\ No newline at end of file
+}
+
+body {
+    margin: 0;
+    font-family: 'KoHo';
+    background-color: #F5F0D7; /* Beige background */
```
Comment (potnurusrilatha) [inline styles/general.css]: i think color should be from variable css

File: styles/general.css [hunk 16]
```diff
@@ -40,4 +40,95 @@ header {
...
+}
+
+/* Side Menu (Full Screen) */
+.side-menu {
+    position: fixed;
+    top: 0;
+    right: 0;
+    width: 100%; /* Full screen width */
+    height: 100%; /* Full screen height */
+    background-color: #F5F0D7; /* Beige background */
```
Comment (potnurusrilatha) [inline styles/general.css]: i think background color should be from texture

**Review by potnurusrilatha — DISMISSED** [review]

**Review by Moon2024-light — COMMENTED** [review]
please check i added variables coding thanks

**Review by Moon2024-light — COMMENTED** [review]
please review again thanks

File: html/reptiles.html [hunk 17]
```diff
@@ -3,9 +3,10 @@
 <head>
     <meta charset="UTF-8">
     <meta name="viewport" content="width=device-width, initial-scale=1.0">
-    <title>Document</title>
+    <title>Linking JavaScript Example</title>
```
Comment (jasminedns) [inline html/reptiles.html]: the title should be reptiles

File: styles/general.css [hunk 18]
```diff
@@ -40,4 +40,95 @@ header {
...
+    z-index: 1000;
+    padding: 10px;
+    border-radius: var(--border-radius);
+    box-shadow: 0px 4px 6px rgba(0, 0, 0, 0.2); /* Shadow effect */
+}
+
+.hamburger-menu .line {
+    width: 30px;
+    height: 4px;
+    background-color: #3C5724; /* Default green */
```
Comment (jasminedns) [inline styles/general.css]: add the variable here

File: styles/general.css [hunk 19]
```diff
@@ -40,4 +40,95 @@ header {
...
+    margin: 6px 0;
+    transition: background-color 0.3s ease;
+}
+
+.hamburger-menu:hover .line {
+    background-color: #F07300; /* Orange on hover */
```
Comment (jasminedns) [inline styles/general.css]: add variable here

File: styles/general.css [hunk 20]
```diff
@@ -40,4 +40,95 @@ header {
...
+    text-align: center;
+}
+
+.side-menu ul li {
+    padding: 20px;
+}
+
+.side-menu ul li a {
+    text-decoration: none;
+    color: #3C5724;
```
Comment (jasminedns) [inline styles/general.css]: add variable

File: styles/general.css [hunk 21]
```diff
@@ -40,4 +40,95 @@ header {
...
+    font-size: 30px;
+    letter-spacing: 2px; /* Slightly spaced letters */
+    display: block;
+    transition: color 0.3s, letter-spacing 0.3s;
+}
+
+.side-menu ul li a:hover {
+    color: #89AA4F; /* Light green on hover */
```
Comment (jasminedns) [inline styles/general.css]: add variable

File: scripts/reptiles.js [hunk 22]
```diff
@@ -0,0 +1,10 @@
+// Ensure the DOM is fully loaded
+document.addEventListener("DOMContentLoaded", () => {
+    // Select the button
+    const button = document.getElementById("myButton");
+    
+    // Add a click event listener to the button
+    button.addEventListener("click", () => {
+        alert("Button was clicked!");
```
Comment (jasminedns) [inline scripts/reptiles.js]: we need the reptiles to appear in the page when clicked, not an alert

File: html/reptiles.html [hunk 23]
```diff
@@ -3,9 +3,10 @@
...
+    <script src="./reptiles.html" defer></script>
 </head>
 <body>
-    
+    <button id="myButton">Click Me!</button>
```
Comment (jasminedns) [inline html/reptiles.html]: You'd probably need 3 since we have 3 reptiles

**Review by jasminedns — CHANGES_REQUESTED** [review]
you could delete reptiles.html and reptiles.js from this branch so we can have the menu ready after the changes. So delete it from this branch and create another one for js so there's less confusion

**Review by Moon2024-light — COMMENTED** [review]
please check changes

**Review by jasminedns — APPROVED** [review]
we have it! good to go imo

**Review by jasminedns — APPROVED** [review]

**Review by TobiasKjernell — APPROVED** [review]

**Pull request merged**